### PR TITLE
PD-6962 middleware chain docs

### PR DESCRIPTION
**Pull request by rvanburen-r7**
Changed verification.md to authenticate.md
Added extra clarification to HTTP-API.md

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* `fetchMetaData(data)`
+  * Fetches meta data about the requesting server for additional verification.
+
+
+* `verifyData(object, version)`
+  * The identity document and data recieved from fetchMetaData will be used to
+  grant or reject a vault token for the server.
+
+
+* `getToken(ttl maybe)`
```
Comment (fmitchell-r7) [inline docs/authentication.md]: Is the `maybe` here a comment about the TTL, or a second argument to the function?

Comment (rvanburen-r7) [inline docs/authentication.md]: The maybe is if we want to specify a TTL in getToken(), or just use a default and remove the TTL argument in getToken().

Comment (fmitchell-r7) [inline docs/authentication.md]: That makes sense. Would it be worth including that in the doc line below? Something like, "`ttl` is optional. If provided, the returned token will have the given TTL. If not provided, the default TTL value of N seconds will be used."

**Comment by fmitchell-r7** [comment]
👍  Thanks for writing this up! :shipit:

**Pull request merged**